// ng2-completer/components/completer-cmp.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-completer/components/completer-cmp';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/forms/src/directives/control_value_accessor';
import * as import10 from '@angular/core/src/linker/view_container';
import * as import11 from '@angular/core/src/security';
import * as import12 from '../../@angular/common/src/directives/ng_if.ngfactory';
import * as import13 from '@angular/core/src/linker/template_ref';
import * as import14 from '@angular/common/src/directives/ng_if';
import * as import15 from 'ng2-completer/components/completer-list-item-cmp';
import * as import16 from './completer-list-item-cmp.ngfactory';
import * as import17 from '../directives/ctr-row.ngfactory';
import * as import18 from '../../@angular/common/src/directives/ng_class.ngfactory';
import * as import19 from '@angular/core/src/linker/element_ref';
import * as import20 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import21 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from 'ng2-completer/directives/ctr-row';
import * as import24 from '../directives/ctr-dropdown.ngfactory';
import * as import25 from '../../@angular/common/src/directives/ng_for.ngfactory';
import * as import26 from '@angular/common/src/directives/ng_for';
import * as import27 from 'ng2-completer/directives/ctr-dropdown';
import * as import28 from '@angular/core/src/linker/query_list';
import * as import29 from '../directives/ctr-completer.ngfactory';
import * as import30 from '../../@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import31 from '../../@angular/forms/src/directives/ng_model.ngfactory';
import * as import32 from '../../@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import33 from '../directives/ctr-input.ngfactory';
import * as import34 from '../directives/ctr-list.ngfactory';
import * as import35 from '@angular/forms/src/directives/default_value_accessor';
import * as import36 from '@angular/forms/src/directives/ng_model';
import * as import37 from '@angular/forms/src/directives/ng_control';
import * as import38 from '@angular/forms/src/directives/ng_control_status';
import * as import39 from 'ng2-completer/directives/ctr-input';
import * as import40 from 'ng2-completer/directives/ctr-list';
import * as import41 from 'ng2-completer/directives/ctr-completer';
export class Wrapper_CompleterCmp {
  /*private*/ _eventHandler:Function;
  context:import0.CompleterCmp;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  constructor() {
    this._changed = false;
    this.context = new import0.CompleterCmp();
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
    this._expr_4 = import1.UNINITIALIZED;
    this._expr_5 = import1.UNINITIALIZED;
    this._expr_6 = import1.UNINITIALIZED;
    this._expr_7 = import1.UNINITIALIZED;
    this._expr_8 = import1.UNINITIALIZED;
    this._expr_9 = import1.UNINITIALIZED;
    this._expr_10 = import1.UNINITIALIZED;
    this._expr_11 = import1.UNINITIALIZED;
    this._expr_12 = import1.UNINITIALIZED;
    this._expr_13 = import1.UNINITIALIZED;
    this._expr_14 = import1.UNINITIALIZED;
    this._expr_15 = import1.UNINITIALIZED;
    this._expr_16 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
  }
  check_dataService(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.dataService = currValue;
      this._expr_0 = currValue;
    }
  }
  check_inputName(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.inputName = currValue;
      this._expr_1 = currValue;
    }
  }
  check_pause(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.pause = currValue;
      this._expr_2 = currValue;
    }
  }
  check_minSearchLength(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.minSearchLength = currValue;
      this._expr_3 = currValue;
    }
  }
  check_maxChars(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.maxChars = currValue;
      this._expr_4 = currValue;
    }
  }
  check_overrideSuggested(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.overrideSuggested = currValue;
      this._expr_5 = currValue;
    }
  }
  check_clearSelected(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.clearSelected = currValue;
      this._expr_6 = currValue;
    }
  }
  check_fillHighlighted(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.fillHighlighted = currValue;
      this._expr_7 = currValue;
    }
  }
  check_placeholder(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.placeholder = currValue;
      this._expr_8 = currValue;
    }
  }
  check_matchClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.matchClass = currValue;
      this._expr_9 = currValue;
    }
  }
  check_textSearching(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_10,currValue))) {
      this._changed = true;
      this.context.textSearching = currValue;
      this._expr_10 = currValue;
    }
  }
  check_textNoResults(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_11,currValue))) {
      this._changed = true;
      this.context.textNoResults = currValue;
      this._expr_11 = currValue;
    }
  }
  check_fieldTabindex(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_12,currValue))) {
      this._changed = true;
      this.context.fieldTabindex = currValue;
      this._expr_12 = currValue;
    }
  }
  check_autoMatch(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_13,currValue))) {
      this._changed = true;
      this.context.autoMatch = currValue;
      this._expr_13 = currValue;
    }
  }
  check_disableInput(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_14,currValue))) {
      this._changed = true;
      this.context.disableInput = currValue;
      this._expr_14 = currValue;
    }
  }
  check_inputClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_15,currValue))) {
      this._changed = true;
      this.context.inputClass = currValue;
      this._expr_15 = currValue;
    }
  }
  check_autofocus(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_16,currValue))) {
      this._changed = true;
      this.context.autofocus = currValue;
      this._expr_16 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.selected.subscribe(_eventHandler.bind(view,'selected'))); }
    if (emit1) { (this.subscription1 = this.context.highlighted.subscribe(_eventHandler.bind(view,'highlighted'))); }
    if (emit2) { (this.subscription2 = this.context.blur.subscribe(_eventHandler.bind(view,'blur'))); }
  }
}
var renderType_CompleterCmp_Host:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.None,([] as any[]),{});
class View_CompleterCmp_Host0 extends import2.AppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.CompleterCmp>;
  _CompleterCmp_0_3:Wrapper_CompleterCmp;
  __NG_VALUE_ACCESSOR_0_4:any[];
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_CompleterCmp_Host0,renderType_CompleterCmp_Host,import6.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_4():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_4 == null)) { (this.__NG_VALUE_ACCESSOR_0_4 = [this._CompleterCmp_0_3.context]); }
    return this.__NG_VALUE_ACCESSOR_0_4;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'ng2-completer',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_CompleterCmp0(this.viewUtils,this,0,this._el_0);
    this._CompleterCmp_0_3 = new Wrapper_CompleterCmp();
    this.compView_0.create(this._CompleterCmp_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._CompleterCmp_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.CompleterCmp) && (0 === requestNodeIndex))) { return this._CompleterCmp_0_3.context; }
    if (((token === import9.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._CompleterCmp_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._CompleterCmp_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const CompleterCmpNgFactory:import8.ComponentFactory<import0.CompleterCmp> = new import8.ComponentFactory<import0.CompleterCmp>('ng2-completer',View_CompleterCmp_Host0,import0.CompleterCmp);
const styles_CompleterCmp:any[] = ['.completer-dropdown[_ngcontent-%COMP%] {\n        border-color: #ececec;\n        border-width: 1px;\n        border-style: solid;\n        border-radius: 2px;\n        width: 250px;\n        padding: 6px;\n        cursor: pointer;\n        z-index: 9999;\n        position: absolute;\n        margin-top: -6px;\n        background-color: #ffffff;\n    }\n\n    .completer-row[_ngcontent-%COMP%] {\n        padding: 5px;\n        color: #000000;\n        margin-bottom: 4px;\n        clear: both;\n        display: inline-block;\n        width: 103%;\n    }\n\n    .completer-selected-row[_ngcontent-%COMP%] {\n        background-color: lightblue;\n        color: #ffffff;\n    }\n\n    .completer-description[_ngcontent-%COMP%] {\n        font-size: 14px;\n    }\n\n    .completer-image-default[_ngcontent-%COMP%] {\n        width: 16px; \n        height: 16px;\n        background-image: url("demo/res/img/default.png");\n    }\n\n    .completer-image-holder[_ngcontent-%COMP%] {\n        float: left;\n        width: 10%;\n    }\n    .completer-item-text-image[_ngcontent-%COMP%] {\n        float: right;\n        width: 90%;\n    }'];
class View_CompleterCmp3 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp3,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','completer-searching'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import3.inlineInterpolate(1,'',this.parentView.parentView.parentView.context.textSearching,'');
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_CompleterCmp4 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp4,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_2 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','completer-no-results'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2:any = import3.inlineInterpolate(1,'',this.parentView.parentView.parentView.context.textNoResults,'');
    if (import3.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_CompleterCmp7 extends import2.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp7,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_1 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'img',new import3.InlineArray2(2,'class','completer-image'),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1:any = import3.inlineInterpolate(1,'',this.parentView.parentView.context.$implicit.image,'');
    if (import3.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'src',this.viewUtils.sanitizer.sanitize(import11.SecurityContext.URL,currVal_1));
      this._expr_1 = currVal_1;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_CompleterCmp8 extends import2.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp8,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','completer-image-default'),(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_CompleterCmp6 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import10.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import12.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import10.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import12.Wrapper_NgIf;
  _text_5:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp6,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','completer-image-holder'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import10.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import13.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import12.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import10.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import13.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import12.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import14.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import13.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import14.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = (this.parentView.context.$implicit.image != '');
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = (this.parentView.context.$implicit.image === '');
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_CompleterCmp7(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_CompleterCmp8(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_CompleterCmp9 extends import2.AppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import15.CompleterListItemCmp>;
  _CompleterListItemCmp_0_3:import16.Wrapper_CompleterListItemCmp;
  _text_1:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp9,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'completer-list-item',new import3.InlineArray2(2,'class','completer-description'),(null as any));
    this.compView_0 = new import16.View_CompleterListItemCmp0(this.viewUtils,this,0,this._el_0);
    this._CompleterListItemCmp_0_3 = new import16.Wrapper_CompleterListItemCmp();
    this._text_1 = this.renderer.createText((null as any),'\n                                ',(null as any));
    this.compView_0.create(this._CompleterListItemCmp_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.CompleterListItemCmp) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) { return this._CompleterListItemCmp_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this.parentView.context.$implicit.description;
    this._CompleterListItemCmp_0_3.check_text(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this.parentView.parentView.parentView.parentView.context.searchStr;
    this._CompleterListItemCmp_0_3.check_searchStr(currVal_0_0_1,throwOnChange,false);
    const currVal_0_0_2:any = this.parentView.parentView.parentView.parentView.context.matchClass;
    this._CompleterListItemCmp_0_3.check_matchClass(currVal_0_0_2,throwOnChange,false);
    const currVal_0_0_3:any = 'description';
    this._CompleterListItemCmp_0_3.check_type(currVal_0_0_3,throwOnChange,false);
    this._CompleterListItemCmp_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_CompleterCmp5 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _CtrRow_2_3:import17.Wrapper_CtrRow;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import10.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import12.Wrapper_NgIf;
  _text_5:any;
  _el_6:any;
  _NgClass_6_3:import18.Wrapper_NgClass;
  _text_7:any;
  _el_8:any;
  compView_8:import2.AppView<import15.CompleterListItemCmp>;
  _CompleterListItemCmp_8_3:import16.Wrapper_CompleterListItemCmp;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _vc_10:import10.ViewContainer;
  _TemplateRef_10_5:any;
  _NgIf_10_6:import12.Wrapper_NgIf;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _map_24:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp5,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_24 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'completer-item-text-image': p0};
    });
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','completer-row-wrapper'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'div',new import3.InlineArray2(2,'class','completer-row'),(null as any));
    this._CtrRow_2_3 = new import17.Wrapper_CtrRow(new import19.ElementRef(this._el_2),this.renderer,(<View_CompleterCmp2>this.parentView)._CtrDropdown_0_3.context);
    this._text_3 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import10.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import13.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import12.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_2,'\n                            ',(null as any));
    this._el_6 = import3.createRenderElement(this.renderer,this._el_2,'div',new import3.InlineArray2(2,'class','completer-item-text'),(null as any));
    this._NgClass_6_3 = new import18.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.parentView.injectorGet(import21.KeyValueDiffers,this.parentView.parentView.parentView.parentIndex),new import19.ElementRef(this._el_6),this.renderer);
    this._text_7 = this.renderer.createText(this._el_6,'\n                                ',(null as any));
    this._el_8 = import3.createRenderElement(this.renderer,this._el_6,'completer-list-item',new import3.InlineArray2(2,'class','completer-title'),(null as any));
    this.compView_8 = new import16.View_CompleterListItemCmp0(this.viewUtils,this,8,this._el_8);
    this._CompleterListItemCmp_8_3 = new import16.Wrapper_CompleterListItemCmp();
    this.compView_8.create(this._CompleterListItemCmp_8_3.context);
    this._text_9 = this.renderer.createText(this._el_6,'\n                                ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor(this._el_6,(null as any));
    this._vc_10 = new import10.ViewContainer(10,6,this,this._anchor_10);
    this._TemplateRef_10_5 = new import13.TemplateRef_(this,10,this._anchor_10);
    this._NgIf_10_6 = new import12.Wrapper_NgIf(this._vc_10.vcRef,this._TemplateRef_10_5);
    this._text_11 = this.renderer.createText(this._el_6,'\n                            ',(null as any));
    this._text_12 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_13 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_2,new import3.InlineArray4(4,'click',(null as any),'mouseenter',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._anchor_10,
      this._text_11,
      this._text_12,
      this._text_13
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import14.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import15.CompleterListItemCmp) && (8 === requestNodeIndex))) { return this._CompleterListItemCmp_8_3.context; }
    if (((token === import13.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import14.NgIf) && (10 === requestNodeIndex))) { return this._NgIf_10_6.context; }
    if (((token === import22.NgClass) && ((6 <= requestNodeIndex) && (requestNodeIndex <= 11)))) { return this._NgClass_6_3.context; }
    if (((token === import23.CtrRow) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._CtrRow_2_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.context.index;
    this._CtrRow_2_3.check_ctrRow(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.context.$implicit;
    this._CtrRow_2_3.check_dataItem(currVal_2_0_1,throwOnChange,false);
    this._CtrRow_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_4_0_0:any = (this.context.$implicit.image || (this.context.$implicit.image === ''));
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_6_0_0:any = 'completer-item-text';
    this._NgClass_6_3.check_klass(currVal_6_0_0,throwOnChange,false);
    const currVal_6_0_1:any = this._map_24((this.context.$implicit.image || (this.context.$implicit.image === '')));
    this._NgClass_6_3.check_ngClass(currVal_6_0_1,throwOnChange,false);
    this._NgClass_6_3.ngDoCheck(this,this._el_6,throwOnChange);
    const currVal_8_0_0:any = this.context.$implicit.title;
    this._CompleterListItemCmp_8_3.check_text(currVal_8_0_0,throwOnChange,false);
    const currVal_8_0_1:any = this.parentView.parentView.parentView.context.searchStr;
    this._CompleterListItemCmp_8_3.check_searchStr(currVal_8_0_1,throwOnChange,false);
    const currVal_8_0_2:any = this.parentView.parentView.parentView.context.matchClass;
    this._CompleterListItemCmp_8_3.check_matchClass(currVal_8_0_2,throwOnChange,false);
    const currVal_8_0_3:any = 'title';
    this._CompleterListItemCmp_8_3.check_type(currVal_8_0_3,throwOnChange,false);
    this._CompleterListItemCmp_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_10_0_0:any = (this.context.$implicit.description && (this.context.$implicit.description != ''));
    this._NgIf_10_6.check_ngIf(currVal_10_0_0,throwOnChange,false);
    this._NgIf_10_6.ngDoCheck(this,this._anchor_10,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_10.detectChangesInNestedViews(throwOnChange);
    this.compView_8.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
    this._vc_10.destroyNestedViews();
    this.compView_8.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_CompleterCmp6(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 10)) { return new View_CompleterCmp9(this.viewUtils,this,10,this._anchor_10,this._vc_10); }
    return (null as any);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._CtrRow_2_3.handleEvent(eventName,$event) && result);
    return result;
  }
}
class View_CompleterCmp2 extends import2.AppView<any> {
  _el_0:any;
  _CtrDropdown_0_3:import24.Wrapper_CtrDropdown;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import10.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import12.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import10.ViewContainer;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import12.Wrapper_NgIf;
  _text_5:any;
  _anchor_6:any;
  /*private*/ _vc_6:import10.ViewContainer;
  _TemplateRef_6_5:any;
  _NgFor_6_6:import25.Wrapper_NgFor;
  _text_7:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp2,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','completer-dropdown','ctrDropdown',''),(null as any));
    this._CtrDropdown_0_3 = new import24.Wrapper_CtrDropdown((<View_CompleterCmp0>this.parentView.parentView)._CtrCompleter_1_3.context,new import19.ElementRef(this._el_0));
    this._text_1 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import10.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import13.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import12.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_4 = new import10.ViewContainer(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import13.TemplateRef_(this,4,this._anchor_4);
    this._NgIf_4_6 = new import12.Wrapper_NgIf(this._vc_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_6 = new import10.ViewContainer(6,0,this,this._anchor_6);
    this._TemplateRef_6_5 = new import13.TemplateRef_(this,6,this._anchor_6);
    this._NgFor_6_6 = new import25.Wrapper_NgFor(this._vc_6.vcRef,this._TemplateRef_6_5,this.parentView.parentView.parentView.injectorGet(import20.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_7 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._anchor_6,
      this._text_7
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import14.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import13.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import14.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    if (((token === import13.TemplateRef) && (6 === requestNodeIndex))) { return this._TemplateRef_6_5; }
    if (((token === import26.NgFor) && (6 === requestNodeIndex))) { return this._NgFor_6_6.context; }
    if (((token === import27.CtrDropdown) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 7)))) { return this._CtrDropdown_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._CtrDropdown_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = (this.parentView.context.searching && this.parentView.parentView.context.displaySearching);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:boolean = (!this.parentView.context.searching && (!this.parentView.context.results || (this.parentView.context.results.length === 0)));
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    const currVal_6_0_0:any = this.parentView.context.results;
    this._NgFor_6_6.check_ngForOf(currVal_6_0_0,throwOnChange,false);
    this._NgFor_6_6.ngDoCheck(this,this._anchor_6,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
    this._vc_6.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
    this._vc_4.destroyNestedViews();
    this._vc_6.destroyNestedViews();
    this._CtrDropdown_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_CompleterCmp3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    if ((nodeIndex == 4)) { return new View_CompleterCmp4(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    if ((nodeIndex == 6)) { return new View_CompleterCmp5(this.viewUtils,this,6,this._anchor_6,this._vc_6); }
    return (null as any);
  }
}
class View_CompleterCmp1 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import10.ViewContainer;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import12.Wrapper_NgIf;
  _text_3:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_CompleterCmp1,renderType_CompleterCmp,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','completer-dropdown-holder'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import10.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import13.TemplateRef_(this,2,this._anchor_2);
    this._NgIf_2_6 = new import12.Wrapper_NgIf(this._vc_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import14.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.context.searchInitialized;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_CompleterCmp2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
var renderType_CompleterCmp:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.Emulated,styles_CompleterCmp,{});
export class View_CompleterCmp0 extends import2.AppView<import0.CompleterCmp> {
  _viewQuery_CtrCompleter_0:import28.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _CtrCompleter_1_3:import29.Wrapper_CtrCompleter;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import18.Wrapper_NgClass;
  _DefaultValueAccessor_3_4:import30.Wrapper_DefaultValueAccessor;
  _NG_VALUE_ACCESSOR_3_5:any[];
  _NgModel_3_6:import31.Wrapper_NgModel;
  _NgControl_3_7:any;
  _NgControlStatus_3_8:import32.Wrapper_NgControlStatus;
  _CtrInput_3_9:import33.Wrapper_CtrInput;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _vc_5:import10.ViewContainer;
  _TemplateRef_5_5:any;
  _CtrList_5_6:import34.Wrapper_CtrList;
  _text_6:any;
  _text_7:any;
  /*private*/ _expr_20:any;
  /*private*/ _expr_21:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  /*private*/ _expr_24:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_CompleterCmp0,renderType_CompleterCmp,import6.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_20 = import1.UNINITIALIZED;
    this._expr_21 = import1.UNINITIALIZED;
    this._expr_22 = import1.UNINITIALIZED;
    this._expr_23 = import1.UNINITIALIZED;
    this._expr_24 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_CtrCompleter_0 = new import28.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'div',new import3.InlineArray4(4,'class','completer-holder','ctrCompleter',''),(null as any));
    this._CtrCompleter_1_3 = new import29.Wrapper_CtrCompleter();
    this._text_2 = this.renderer.createText(this._el_1,'\n            ',(null as any));
    this._el_3 = import3.createRenderElement(this.renderer,this._el_1,'input',new import3.InlineArray16(12,'autocapitalize','off','autocomplete','off','autocorrect','off','class','completer-input','ctrInput','','type','search'),(null as any));
    this._NgClass_3_3 = new import18.Wrapper_NgClass(this.parentView.injectorGet(import20.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import21.KeyValueDiffers,this.parentIndex),new import19.ElementRef(this._el_3),this.renderer);
    this._DefaultValueAccessor_3_4 = new import30.Wrapper_DefaultValueAccessor(this.renderer,new import19.ElementRef(this._el_3));
    this._NG_VALUE_ACCESSOR_3_5 = [this._DefaultValueAccessor_3_4.context];
    this._NgModel_3_6 = new import31.Wrapper_NgModel((null as any),(null as any),(null as any),this._NG_VALUE_ACCESSOR_3_5);
    this._NgControl_3_7 = this._NgModel_3_6.context;
    this._NgControlStatus_3_8 = new import32.Wrapper_NgControlStatus(this._NgControl_3_7);
    this._CtrInput_3_9 = new import33.Wrapper_CtrInput(this._CtrCompleter_1_3.context,this._NgModel_3_6.context);
    this._text_4 = this.renderer.createText(this._el_1,'\n\n            ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._vc_5 = new import10.ViewContainer(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import13.TemplateRef_(this,5,this._anchor_5);
    this._CtrList_5_6 = new import34.Wrapper_CtrList(this._CtrCompleter_1_3.context,this._TemplateRef_5_5,this._vc_5.vcRef,this.ref);
    this._text_6 = this.renderer.createText(this._el_1,'\n        ',(null as any));
    this._text_7 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_3,new import3.InlineArray16(10,'ngModelChange',(null as any),'blur',(null as any),'input',(null as any),'keyup',(null as any),'keydown',(null as any)),this.eventHandler(this.handleEvent_3));
    this._NgModel_3_6.subscribe(this,this.eventHandler(this.handleEvent_3),true);
    this._CtrInput_3_9.subscribe(this,this.eventHandler(this.handleEvent_3),true);
    this._viewQuery_CtrCompleter_0.reset([this._CtrCompleter_1_3.context]);
    this.context.completer = this._viewQuery_CtrCompleter_0.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._text_7
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (3 === requestNodeIndex))) { return this._NgClass_3_3.context; }
    if (((token === import35.DefaultValueAccessor) && (3 === requestNodeIndex))) { return this._DefaultValueAccessor_3_4.context; }
    if (((token === import9.NG_VALUE_ACCESSOR) && (3 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_3_5; }
    if (((token === import36.NgModel) && (3 === requestNodeIndex))) { return this._NgModel_3_6.context; }
    if (((token === import37.NgControl) && (3 === requestNodeIndex))) { return this._NgControl_3_7; }
    if (((token === import38.NgControlStatus) && (3 === requestNodeIndex))) { return this._NgControlStatus_3_8.context; }
    if (((token === import39.CtrInput) && (3 === requestNodeIndex))) { return this._CtrInput_3_9.context; }
    if (((token === import13.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import40.CtrList) && (5 === requestNodeIndex))) { return this._CtrList_5_6.context; }
    if (((token === import41.CtrCompleter) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._CtrCompleter_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._CtrCompleter_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = 'completer-input';
    this._NgClass_3_3.check_klass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this.context.inputClass;
    this._NgClass_3_3.check_ngClass(currVal_3_0_1,throwOnChange,false);
    this._NgClass_3_3.ngDoCheck(this,this._el_3,throwOnChange);
    this._DefaultValueAccessor_3_4.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_3_2_0:any = this.context.disableInput;
    this._NgModel_3_6.check_isDisabled(currVal_3_2_0,throwOnChange,false);
    const currVal_3_2_1:any = this.context.searchStr;
    this._NgModel_3_6.check_model(currVal_3_2_1,throwOnChange,false);
    this._NgModel_3_6.ngDoCheck(this,this._el_3,throwOnChange);
    this._NgControlStatus_3_8.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_3_4_0:any = this.context.clearSelected;
    this._CtrInput_3_9.check_clearSelected(currVal_3_4_0,throwOnChange,false);
    const currVal_3_4_1:any = this.context.overrideSuggested;
    this._CtrInput_3_9.check_overrideSuggested(currVal_3_4_1,throwOnChange,false);
    const currVal_3_4_2:any = this.context.fillHighlighted;
    this._CtrInput_3_9.check_fillHighlighted(currVal_3_4_2,throwOnChange,false);
    this._CtrInput_3_9.ngDoCheck(this,this._el_3,throwOnChange);
    const currVal_5_0_0:any = this.context.minSearchLength;
    this._CtrList_5_6.check_ctrListMinSearchLength(currVal_5_0_0,throwOnChange,false);
    const currVal_5_0_1:any = this.context.pause;
    this._CtrList_5_6.check_ctrListPause(currVal_5_0_1,throwOnChange,false);
    const currVal_5_0_2:any = this.context.autoMatch;
    this._CtrList_5_6.check_ctrListAutoMatch(currVal_5_0_2,throwOnChange,false);
    const currVal_5_0_3:any = this.context.dataService;
    this._CtrList_5_6.check_dataService(currVal_5_0_3,throwOnChange,false);
    this._CtrList_5_6.ngDoCheck(this,this._anchor_5,throwOnChange);
    this._vc_5.detectChangesInNestedViews(throwOnChange);
    const currVal_20:any = this.context.inputName;
    if (import3.checkBinding(throwOnChange,this._expr_20,currVal_20)) {
      this.renderer.setElementAttribute(this._el_3,'name',((currVal_20 == null)? (null as any): currVal_20.toString()));
      this._expr_20 = currVal_20;
    }
    const currVal_21:any = this.context.placeholder;
    if (import3.checkBinding(throwOnChange,this._expr_21,currVal_21)) {
      this.renderer.setElementProperty(this._el_3,'placeholder',currVal_21);
      this._expr_21 = currVal_21;
    }
    const currVal_22:any = this.context.maxChars;
    if (import3.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this.renderer.setElementAttribute(this._el_3,'maxlength',((currVal_22 == null)? (null as any): currVal_22.toString()));
      this._expr_22 = currVal_22;
    }
    const currVal_23:any = this.context.fieldTabindex;
    if (import3.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setElementProperty(this._el_3,'tabIndex',currVal_23);
      this._expr_23 = currVal_23;
    }
    const currVal_24:any = this.context.autofocus;
    if (import3.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setElementProperty(this._el_3,'autofocus',currVal_24);
      this._expr_24 = currVal_24;
    }
    this._NgControlStatus_3_8.checkHost(this,this,this._el_3,throwOnChange);
  }
  destroyInternal():void {
    this._vc_5.destroyNestedViews();
    this._NgModel_3_6.ngOnDestroy();
    this._CtrInput_3_9.ngOnDestroy();
    this._CtrCompleter_1_3.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 5)) { return new View_CompleterCmp1(this.viewUtils,this,5,this._anchor_5,this._vc_5); }
    return (null as any);
  }
  handleEvent_3(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._DefaultValueAccessor_3_4.handleEvent(eventName,$event) && result);
    result = (this._CtrInput_3_9.handleEvent(eventName,$event) && result);
    if ((eventName == 'ngModelChange')) {
      const pd_sub_0:any = ((<any>(this.context.searchStr = $event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'ngModelChange')) {
      const pd_sub_1:any = ((<any>this.context.onChange($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.context.onBlur()) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
}
